### PR TITLE
Fixes #5666 - Default options for transformations

### LINKED ISSUE #5666: Default Options for Transformations
Hi,

It'd be nice if default options could be set in the 
conf file for all transformations.

---
- **Original URL**: "https://sourceforge.net/p/phpmyadmin/feature-requests/967/":https://sourceforge.net/p/phpmyadmin/feature-requests/967/
- **Original author**: olafvdspek

### PR DESCRIPTION
**Pull request by Achilles-96**
Signed-off-by: Raghuram Vadapalli <raghuram.vadapalli@research.iiit.ac.in>

Before submitting pull request, please check that every commit:

- [x] Has proper Signed-Off-By
- [x] Has commit message which describes it
- [x] Is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any new functionality is covered by tests

### CODE REVIEW

**Comment by Achilles-96** [comment]
GSoC task 2: 
I thought I will have to add this for all transformations in `PMA_HOME/libraries/plugins/transformations/abs/`, but none of the others use default options (or any options). So I added only for these five transformations. Please correct me if I am wrong.

**Comment by ibennetch** [comment]
I don't know why the bot thinks your commits don't have the Signed-off-by line. This is very strange; it happened in your other pull request as well, even though I can see it as part of the commit message.

**Comment by Achilles-96** [comment]
@ibennetch , I see that there are many transformations don't make use of `$options`. Do you think it is a better idea to still have an empty array as default.

File: libraries/config.default.php [hunk 1]
```diff
@@ -3091,3 +3091,78 @@
...
+
+/**
+ * Default transformations for hex
+ *
+ * @global array $cfg['DefaultTransformations']
+ */
+$cfg['DefaultTransformations'] = array();
+
+/**
+ * Default transformations for hex
```
Comment (ibennetch) [inline libraries/config.default.php]: Looks like this comment comes from copying and pasting the one above. This probably should be "Default transformations for substring", right?

File: libraries/config.default.php [hunk 2]
```diff
@@ -3091,3 +3091,78 @@
...
  */
 $cfg['DisableShortcutKeys'] = false;
+
+
+/*******************************************************************************
+ * Default options for transformations
+ */
+
+/**
+ * Default transformations for hex
```
Comment (ibennetch) [inline libraries/config.default.php]: This comment probably should be something like "Initialize default transformation array", although I suppose that's kind of obvious from the code.

**Comment by ibennetch** [comment]
This looks fine to me and works well. I restarted the tests and will wait for them to finish before merging.

**Comment by ibennetch** [comment]
We should also add a section about this to the documentation; at least mention it in the transformation portion and list all the directives in the configuration section.

File: libraries/config/setup.forms.php [hunk 3]
```diff
@@ -211,7 +211,11 @@
     'ShowDbStructureLastCheck');
 $forms['Main_panel']['TableStructure'] = array(
     'HideStructureActions',
-    'ShowColumnComments');
+    'ShowColumnComments',
+    'DefaultTransformations' => array(
+        'Hex'
+    ));
```
Comment (ibennetch) [inline libraries/config/setup.forms.php]: This should just be `)`

**Comment by ibennetch** [comment]
From the test suite (Travis), I see these test problems:

```
There were 2 errors:

1) TransformationPluginsTest::testTransformation with data set #28 (PhpMyAdmin\Plugins\Transformations\Output\Image_JPEG_Inline Object (), array('PMA_JPEG_Inline', array('./image/', '200', 'PMA_wrapper_link')), '<a href="transformation_wrapp.../></a>')

Undefined index: wrapper_link

/home/travis/build/phpmyadmin/phpmyadmin/libraries/classes/Plugins/Transformations/Abs/InlineTransformationsPlugin.php:53

/home/travis/build/phpmyadmin/phpmyadmin/test/classes/plugin/transformations/TransformationPluginsTest.php:999

2) TransformationPluginsTest::testTransformation with data set #29 (PhpMyAdmin\Plugins\Transformations\Output\Image_PNG_Inline Object (), array('PMA_PNG_Inline', array('./image/', '200', 'PMA_wrapper_link')), '<a href="transformation_wrapp.../></a>')

Undefined index: wrapper_link

/home/travis/build/phpmyadmin/phpmyadmin/libraries/classes/Plugins/Transformations/Abs/InlineTransformationsPlugin.php:53

/home/travis/build/phpmyadmin/phpmyadmin/test/classes/plugin/transformations/TransformationPluginsTest.php:999

--

There was 1 failure:

1) TransformationPluginsTest::testTransformation with data set #22 (PhpMyAdmin\Plugins\Transformations\Text_Plain_Link Object (), array('https://example.com/PMA_TXT_LINK', array()), '<a href="https://example.com/...NK</a>')

Failed asserting that two strings are equal.

--- Expected

+++ Actual

@@ @@

-'<a href="https://example.com/PMA_TXT_LINK" title="" target="_blank" rel="noopener noreferrer">https://example.com/PMA_TXT_LINK</a>'

+'<a href="https://example.com/PMA_TXT_LINK" title="" target="_blank" rel="noopener noreferrer"></a>'

/home/travis/build/phpmyadmin/phpmyadmin/test/classes/plugin/transformations/TransformationPluginsTest.php:999
```

**Comment by ibennetch** [comment]
The test is correct in that I have a column of type varchar with content `https://example.com` and I've applied the TextLink transformation. Before applying your code, it appears as expected, but after your commits, the transformed output is blank.

**Comment by ibennetch** [comment]
Great! Good work.

**Pull request merged**